Properly show case importer errors to user

dannyroberts opened the pull request:
##### SUMMARY
This PR is another in what seems to be a chain of changes over years, the last of which was https://github.com/dimagi/commcare-hq/pull/23533.

<img width="1194" alt="Screen Shot 2020-02-03 at 7 51 52 PM" src="https://user-images.githubusercontent.com/137212/73707009-c53c9080-46bf-11ea-9903-ea77f7385541.png">

dannyroberts committed:
Suppress and report errors fetching case importer task

dannyroberts committed:
Properly show case importer errors to user

Rohit25negi commented:
addressing icds test case failure here: https://github.com/dimagi/commcare-hq/pull/26552, please merge the other PR before this.

dannyroberts committed:
Merge branch 'master' into dmr/case-importer-errors

dannyroberts committed:
Fix test after case importer error fix

by using a memoized exception that is guaranteed to be the same object
in the code and in the test

millerdev commented on corehq/apps/case_importer/tasks.py:
```diff
@@ -24,7 +24,7 @@ def bulk_import_async(config, domain, excel_id):
         case_upload.check_file()
     except ImporterError as e:
-        update_task_state(bulk_import_async, states.FAILURE, {'errors': get_importer_error_message(e)})
+        update_task_state(bulk_import_async, states.FAILURE, Exception(get_importer_error_message(e)))
```
Rather than using `Exception`, which can only be caught with `except Exception` (and that catches almost every other error too), would it make sense to use a more limited error type for these?

I'm assuming this error is raised somewhere, but I can't see it happens in this PR. Maybe my comment doesn't even make sense?

dannyroberts replied:
The error is actually _not_ ever raised anywhere. Celery just requires that the result for a FAILURE be either an exception or a very specifically formatted dict, and I thought using `Exception` would highlight that it's just filling in for something of type Exception

dannyroberts committed:
Pull out exit_celery_with_error_message util in case importer
